packages/cloud_firestore/cloud_firestore/lib/src/firestore.dart

```diff
@@ -35,16 +37,25 @@ class FirebaseFirestore extends FirebasePluginPlatform {
   /// Returns an instance using a specified [FirebaseApp].
   static FirebaseFirestore instanceFor({
     required FirebaseApp app,
+    @Deprecated(
+      '`databaseURL` has been deprecated. Please use `databaseId` instead.',
+    )
     String? databaseURL,
+    String? databaseId,
   }) {
-    String url = databaseURL ?? '(default)';
-    String cacheKey = '${app.name}|$url';
+    String firestoreDatabaseId = databaseId ?? databaseURL ?? '(default)';
+    String cacheKey = '${app.name}|$firestoreDatabaseId';
     if (_cachedInstances.containsKey(cacheKey)) {
       return _cachedInstances[cacheKey]!;
     }
 
     FirebaseFirestore newInstance =
-        FirebaseFirestore._(app: app, databaseURL: url);
+        // Both databaseURL and databaseId are required so we have to pass both for now. We can remove databaseURL in a future release.
+        FirebaseFirestore._(
+      app: app,
+      databaseURL: firestoreDatabaseId,
```

Comment:
Couldn't we remove it here since it's a private constructor, and won't be a breaking change for the user?

Reply:
We could but we would then have to initialise it in a constructor function. It is one or the other so it doesn't make a difference.